packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -276,26 +283,38 @@ Future<LaunchResult> _startApp(DriveCommand command) async {
 }
 
 /// Runs driver tests.
-typedef TestRunner = Future<void> Function(List<String> testArgs, String observatoryUri);
+typedef TestRunner = Future<void> Function(List<String> testArgs, String observatoryUri, bool useTestCore);
 TestRunner testRunner = _runTests;
 void restoreTestRunner() {
   testRunner = _runTests;
 }
 
-Future<void> _runTests(List<String> testArgs, String observatoryUri) async {
+Future<void> _runTests(List<String> testArgs, String observatoryUri, bool useTestCore) async {
   printTrace('Running driver tests.');
 
   PackageMap.globalPackagesPath = fs.path.normalize(fs.path.absolute(PackageMap.globalPackagesPath));
   final String dartVmPath = fs.path.join(dartSdkPath, 'bin', 'dart');
   final String testScript = fs.path.join(Cache.flutterRoot, 'packages', 'flutter_tools', 'lib', 'src', 'test', 'drive_executable.dart');
-  final int result = await runCommandAndStreamOutput(
-    <String>[dartVmPath]
-      ..addAll(dartVmFlags)
-      ..addAll(<String>['--packages=${PackageMap.globalPackagesPath}', testScript])
-      ..addAll(testArgs)
-      ..add('-rexpanded'),
-    environment: <String, String>{ 'VM_SERVICE_URL': observatoryUri }
-  );
+  int result;
+  if (useTestCore) {
+    result = await runCommandAndStreamOutput(
+      <String>[dartVmPath]
+        ..addAll(dartVmFlags)
+        ..addAll(<String>['--packages=${PackageMap.globalPackagesPath}', testScript])
+        ..addAll(testArgs)
+        ..add('-rexpanded'),
+      environment: <String, String>{ 'VM_SERVICE_URL': observatoryUri });
+  } else {
+    result = await runCommandAndStreamOutput(
+      <String>[dartVmPath]
+        ..addAll(dartVmFlags)
+        ..addAll(testArgs)
+        ..addAll(<String>[
+          '--packages=${PackageMap.globalPackagesPath}',
+           '-rexpanded',
+        ]),
+      environment: <String, String>{ 'VM_SERVICE_URL': observatoryUri });
+  }
```

Comment:
nit: I thin you could reduce some code repetition across these `if`/`else` blocks.